src/ChluIPFS.js

```diff
@@ -167,37 +167,98 @@ class ChluIPFS {
+        if (reviewRecord.orbitDb) {
+            const db = await this.openDbForReplication(reviewRecord.orbitDb);
+            db.events.once('replicated', () => this.notifyIfReviewIsUpdated(db, multihash, notifyUpdate));
+            this.notifyIfReviewIsUpdated(db, multihash, notifyUpdate);
+        }
+    }
+
+    async getReviewRecord(multihash){
         const dagNode = await this.ipfs.object.get(this.utils.multihashToBuffer(multihash));
-        const buffer = dagNode.Data;
+        const buffer = dagNode.data;
         const messages = protons(require('../src/utils/protobuf'));
         const reviewRecord = messages.ReviewRecord.decode(buffer);
-        // TODO: validate
         return reviewRecord;
     }
 
-    async storeReviewRecord(reviewRecord){
+    async readReviewRecord(multihash, notifyUpdate = null) {
+        const reviewRecord = await this.getReviewRecord(multihash);
+        // TODO validate
+        if (notifyUpdate) this.findLastReviewRecordUpdate(multihash, notifyUpdate);
+        return reviewRecord;
+    }
+
+    async storeReviewRecord(reviewRecord, previousVersionMultihash = null){
         if (this.type !== constants.types.customer) {
             throw new Error('Not a customer');
         }
-        let buffer;
-        if (Buffer.isBuffer(reviewRecord)) {
-            buffer = reviewRecord;
-        } else if (typeof reviewRecord === 'object') {
-            const messages = protons(require('./utils/protobuf'));
-            buffer = messages.ReviewRecord.encode(reviewRecord);
-        } else {
-            throw new Error('Unrecognised reviewRecord type: either pass a protobuf encoded buffer or an object');
-        }
+        reviewRecord.orbitDb = this.getOrbitDBAddress();
+        const messages = protons(require('./utils/protobuf'));
+        const buffer = messages.ReviewRecord.encode(reviewRecord);
+        // TODO validate
         // write thing to ipfs
         const dagNode = await this.ipfs.object.put(buffer);
         const multihash = this.utils.multihashToString(dagNode.multihash);
         // Broadcast request for pin, then wait for response
         // TODO: handle a timeout and also rebroadcast periodically, otherwise new peers won't see the message
-        await new Promise(fullfill => {
+        let tasksToAwait = [];
+        tasksToAwait.push(new Promise(fullfill => {
             this.events.once(constants.eventTypes.pinned + '_' + multihash, () => fullfill());
             this.room.broadcast(this.utils.encodeMessage({ type: constants.eventTypes.wroteReviewRecord, multihash }));
+        }));
+        if (previousVersionMultihash) {
+            // This is a review update
+            tasksToAwait.push(this.setForwardPointerForReviewRecord(previousVersionMultihash, multihash));
+        }
+        await Promise.all(tasksToAwait);
+        return multihash;
+    }
+
+    async setForwardPointerForReviewRecord(previousVersionMultihash, multihash) {
```

Comment:
@fazo96 do we need to set the fwd pointer? iirc, we agreed to leave this as it is an optimisation, and not required for correctness of the protocol. Do I remember correctly?

Reply:
This function is required if you need to find an updated version of a review given the multihash of a version that is not the latest (which is the whole point of this PR).

We could remove this but then there is no way to get the latest version of a review so we have to figure out another solution.

However if we remove this function (and the ability to find updated versions of a review using this system) we would also remove orbit-db since we are not using it for anything else right now.

I think we both agreed to different things when we talked the last time, I'll make a better effort at clarity and writing down the conclusion next time.